### PR TITLE
fix: update hardhat scripts to use getContractAt

### PR DESCRIPTION
**Pull request by 0xGrapefruit**

### CODE REVIEW

File: hardhat.tasks.js [hunk 1]
```diff
@@ -20,37 +18,20 @@ task("tenderly-verify", "verifies current deployment on tenderly").setAction(
...
-    const { state } = args;
-
-    const currentState = await podEnsRegistrar.state();
-    if (currentState === stateEnum[state]) {
-      console.log("Contract was already set to that state");
-      return;
-    }
-    await podEnsRegistrar.setRestrictionState(stateEnum[state]);
-    console.log(`Successfully changed state to ${state}`);
-  });
```
Comment (kevinweaver) [inline hardhat.tasks.js:45]: ❓ Did we decide we no longer need to maintain this state?

**Review by kevinweaver — COMMENTED** [review]
❓ What's the benefit of using hardhat-ethers `getContractAtFromArtifact` over ethers `getContract`? Something about already having the artifacts locally?

File: hardhat.tasks.js (hunk 1 above)
Comment (jtlacci) [reply hardhat.tasks.js:45]: yah, we are no longer using the $SHIP token gating, the only thing we're gating is frontend via the permit NFTs

**Comment by kevinweaver** [comment]
> `getContract` was something that was passed to the Hardhat Runtime Enviornment (HRE) by waffle which was a mocking plugin that they deperacted in favor of a first party implementation. Was honestly very convenient, but potentially abstracting away where we are getting the contracts from is an antipattern

Ah that makes sense, thanks!

File: hardhat.tasks.js [hunk 2]
```diff
@@ -117,42 +97,53 @@ task("register-controller", "registers controller with controller registry")
...
     const { ensHolder } = await getNamedAccounts();
     const ensHolderSigner = ethers.provider.getSigner(ensHolder);
 
-    const network = await getChainId();
-    const ensRegistryAddress =
-      network === "31337" ? (await deployments.get("ENSRegistry")).address : getEnsAddress(network);
+    const ensRegistry = await ethers.getContractAtFromArtifact(
+      await artifacts.readArtifact("ENSRegistry"),
+      ensHolderSigner,
+    );
```
Comment (kevinweaver) [inline hardhat.tasks.js:107]: ❓ We're doing something similar on lines 151 and 311, should we remove the network checks in those cases too?

Comment (0xGrapefruit) [reply hardhat.tasks.js:107]: oops good catch, we actually need those checks.

**Review by kevinweaver — APPROVED** [review]
👍 Looks good

**Review by jtlacci — APPROVED** [review]

**Pull request merged**